File: tests/fixtures/check_then_act/py/continue_guard_fixture.py
# A guard that exits by continue covers the rest of its loop, not the code after the loop.
async def drain(repo, ids):
    for item_id in ids:
        value = repo.get(item_id)
        if not value:  # expect: check
            continue
        await repo.sync(item_id)  # expect: gap await
        repo.update(item_id)  # expect: write


async def after_loop(repo, ids):
    for item_id in ids:
        value = repo.get(item_id)
        if not value:
            continue
    await repo.flush()
    repo.update('all')

File: tests/fixtures/check_then_act/ts/continue_guard_fixture.ts
// A guard that exits by continue covers the rest of its loop, not the code after the loop.
export async function drain(repo: any, ids: string[]) {
  for (const id of ids) {
    const value = repo.get(id);
    if (!value) continue; // expect: check
    await repo.sync(id); // expect: gap await
    repo.update(id); // expect: write
  }
}

export async function afterLoop(repo: any, ids: string[]) {
  for (const id of ids) {
    const value = repo.get(id);
    if (!value) continue;
  }
  await repo.flush();
  repo.update('all');
}
